### PR TITLE
Reduce profile size and do not export nil values

### PR DESCRIPTION
**Pull request by imobachgs**
This PR implements these changes:

* Do not export nil values (fix for [bsc#1172822](https://bugzilla.suse.com/show_bug.cgi?id=1172822)).
* Do no export any setting if ypbind is not even installed ([bsc#1172749](https://bugzilla.opensuse.org/show_bug.cgi?id=1172749)).

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+Metrics/ModuleLength:
+  Max: 1400
+
+# Offense count: 2
+# Configuration parameters: CountKeywordArgs.
+Metrics/ParameterLists:
+  Max: 7
+
+# Offense count: 13
+Lint/UselessAssignment:
```
Comment (jreidinger) [inline .rubocop.yml:47]: why is this disabled? This usually reduce dead code

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+Naming/UncommunicativeMethodParamName:
+  Exclude:
+    - 'src/include/nis/ui.rb'
+    - 'src/modules/Nis.rb'
+
+# Offense count: 1
+# Cop supports --auto-correct.
+# Configuration parameters: Keywords.
+# Keywords: TODO, FIXME, OPTIMIZE, HACK, REVIEW
+Style/CommentAnnotation:
```
Comment (jreidinger) [inline .rubocop.yml:106]: I think this is also not needed, especially when it can be auto corrected.

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  Exclude:
+    - 'src/clients/nis.rb'
+
+# Offense count: 1
+Style/CommentedKeyword:
```
Comment (jreidinger) [inline .rubocop.yml:111]: this is also quite tricky. Does it really provide value?

File: .rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    - 'src/include/nis/ui.rb'
+    - 'src/modules/Nis.rb'
+
+# Offense count: 1
+Style/MultipleComparison:
+  Exclude:
+    - 'src/include/nis/ui.rb'
+
+# Offense count: 1
+Style/NestedTernaryOperator:
```
Comment (jreidinger) [inline .rubocop.yml:153]: Ah, nested ternary is really ugly :) But it is up to you.

File: src/modules/Nis.rb [hunk 5]
```diff
@@ -753,6 +758,7 @@ def Export
         "slp_domain"        => @slp_domain,
         "netconfig_policy"  => @policy
       }
+      settings.reject { |_k, v| v.nil? }
```
Comment (jreidinger) [inline src/modules/Nis.rb:761]: I worry, this is a bit cheap fix :) . Options default is "", so nil has to be assigned, and I think that is root of issue. options should be always String.

Comment (imobachgs) [reply src/modules/Nis.rb]: I was unsure about how to solve it, but I think you are right. I found out that the problem is the initialization when reading from the configuration file. Fixed now.

File: src/modules/Nis.rb [hunk 6]
```diff
@@ -730,6 +733,8 @@ def Import(settings)
     # Dump the NIS settings to a map, for autoinstallation use.
     # @return $["start":, "servers":[...], "domain":]
     def Export
+      return {} unless Yast::Package.Installed(NIS_CLIENT_PACKAGE)
```
Comment (teclator) [inline src/modules/Nis.rb:736]: Do not have consequences when Import and Export? I was wondering if could affect AutoYaST config for example.

Comment (imobachgs) [reply src/modules/Nis.rb:736]: That's the idea. Do not export anything when the package is not installed.

Comment (imobachgs) [reply src/modules/Nis.rb]: Oh, now I get it. You are right, I have fixed the problem.

**Pull request merged**